tests/quantization/autoawq/test_awq.py

```diff
@@ -104,15 +107,41 @@ def test_quantized_model_conversion(self):
         """
         from awq.modules.linear import WQLinear_GEMM, WQLinear_GEMV
 
-        self.tokenizer(self.input_text, return_tensors="pt").to(torch_device)
+        from transformers.integrations.awq import replace_with_awq_linear
+
+        model_id = "facebook/opt-350m"
+        config = AutoConfig.from_pretrained(model_id, revision="cb32f77e905cccbca1d970436fb0f5e6b58ee3c5")
+        quantization_config = AwqConfig(bits=4)
+
+        with init_empty_weights():
+            model = OPTForCausalLM(config)
```

Comment:
Do we need to use `OPTForCausalLM` specifically here. Why not `AutoModelForCausalLM`?

Reply:
Because initializing a model with a config file through `AutoModelForCausalLM` is not supported:

```bash
>>> AutoModelForCausalLM(config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/younes_huggingface_co/code/transformers/src/transformers/models/auto/auto_factory.py", line 411, in __init__
    raise EnvironmentError(
OSError: AutoModelForCausalLM is designed to be instantiated using the `AutoModelForCausalLM.from_pretrained(pretrained_model_name_or_path)` or `AutoModelForCausalLM.from_config(config)` methods.
```